runtime/src/parachains.rs

```diff
@@ -431,6 +384,11 @@ impl<T: Trait> Module<T> {
 
 		let mut ingress_update = BTreeMap::new();
 
+		// we sort them in order to provide a fast lookup to ensure we can avoid duplicates in the
+		// needs_dispatch queue.
+		let mut ordered_needs_dispatch = NeedsDispatch::get();
+		ordered_needs_dispatch.sort_unstable();
```

Comment:
the documentation on `NeedsDispatch` says that it is sorted - either it is wrong or this line is redundant

Reply:
they should be sorted, but since there's no way of proving that here, this is a defensive measure to ensure things don't needlessly break if for whatever reason state is corrupted.

Reply:
yeah it's redundant. removing.